File: JavaScript/leetcode_226.js
/* 
Given the root of a binary tree, invert the tree, and return its root.
*/

var invertTree = function (root) {
    if (!root) return null;

    // Swap left and right
    [root.left, root.right] = [root.right, root.left];

    // Recursively invert subtrees
    invertTree(root.left);
    invertTree(root.right);

    return root;
};

File: Python/leetcode_226.py
'''
Given the root of a binary tree, invert the tree, and return its root.
'''

from typing import Optional


class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right


class Solution:
    def invert_tree(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        if not root:
            return None

        # Swap left and right
        root.left, root.right = root.right, root.left

        # Recursively invert subtrees
        self.invert_tree(root.left)
        self.invert_tree(root.right)

        return root
